### PR TITLE
Boilerplate website

### LINKED ISSUE #26: Develop React Boilerplate Website
Currently the boilerplate has a very naive website : http://react-simple-boilerplate.surge.sh/

Add all the components and proper documentation in the website with proper SEO.

### PR DESCRIPTION
**Pull request by akshay-nm**
(old PR #59)
Resolves #26, #77  

### Checklist

- [x] Initialize project using create-next-app in `/website` directory
- [x] Add home page and its components
- [x] Add Docs page and its components

#### Additional information
- Fixed a lint error in a button component in boilerplate which was causing CI issues.

### CODE REVIEW

**Comment by mundane140** [comment]
@udaypydi there might be something I am missing here. `yarn workspace` command does not detect the `react-simple-boilerplate` workspace. When I run `yarn workspaces info` it shows two workspaces `react-simple-boilerplate` and `website`.

**Comment by udaypydi** [comment]
@mundane140 can you please fix the shields?
[Shields](https://travis-ci.org/github/udaypydi/react-simple-boilerplate/builds/702146021)

**Comment by udaypydi** [comment]
> @udaypydi there might be something I am missing here. `yarn workspace` command does not detect the `react-simple-boilerplate` workspace. When I run `yarn workspaces info` it shows two workspaces `react-simple-boilerplate` and `website`.

 @mundane140  Let's use lerna. That would give more flexibility in maintaining versions and publishing packages as well.

**Comment by mundane140** [comment]
> @mundane140 can you please fix the shields?
> [Shields](https://travis-ci.org/github/udaypydi/react-simple-boilerplate/builds/702146021)

Please elaborate.

**Comment by udaypydi** [comment]
@akshay-nm   can you please fix this `https://travis-ci.org/github/udaypydi/react-simple-boilerplate/builds/702146021` and also resolve the conflict.

**Comment by akshay-nm** [comment]
@udaypydi I have configured the project as Lerna monorepo. Lerna is now a dev dependency in the root `package.json`.
Requesting review.

File: package.json [hunk 1]
```diff
@@ -1,75 +1,16 @@
...
-  "keywords": [
-    "react",
-    "boilerplate",
-    "webpack",
-    "webpack hmr",
-    "travis",
-    "react-simple-boilerplate",
-    "react-boilerplate"
+  "private": true,
+  "workspaces": [
```
Comment (udaypydi) [inline package.json:3]: Let’s add specific packages. Instead of ‘packages/*’

Comment (udaypydi) [reply package.json:3]: In this case it’s boilerplate and website

**Comment by udaypydi** [comment]
@akshay-nm please resolve conflicts.

**Comment by akshay-nm** [comment]
@udaypydi Done

File: package.json (hunk 1 above)
Comment (akshay-nm) [reply package.json:3]: I tried was experiencing issues while running root level scripts based on lerna. 
Moreover, that is probably not related to the scope of this PR.

Comment (akshay-nm) [reply package.json:3]: > In this case it’s boilerplate and website

For this you might also need to rename the boilerplate package in its package.json

**Comment by akshay-nm** [comment]
A gentle reminder @udaypydi 😅
Let's merge this before we start having merge conflicts.

**Comment by udaypydi** [comment]
> A gentle reminder @udaypydi 😅
> Let's merge this before we start having merge conflicts.

Yep I was bit busy with other OS projects and PRs. Will review it tonight and will merge it.

File: packages/website/docs/supported-browsers-and-features.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+topic: "Getting Started"
+title: "Supported Borwsers and Features"
+index: "3"
+date: "2020-01-02"
+---
+
+Pariatur officia ullamco et do incididunt pariatur fugiat proident Lorem. Cupidatat commodo amet ad consequat deserunt esse proident do velit amet ad tempor esse. Ut irure aliquip labore ex adipisicing ullamco id. Sit labore nostrud deserunt aliquip nostrud nulla. Dolore ad cupidatat cillum enim proident consectetur qui quis in non enim proident.
```
Comment (udaypydi) [inline packages/website/docs/supported-browsers-and-features.md:8]: is this Spanish ? Maybe... 🤔

**Comment by udaypydi** [comment]
@mvhk Please pull this branch and test out the website as well.  I am held up with other projects. I will review the code and merge it if things are working fine. 

@akshay-nm Can you please update a loom file of how it looks like.

File: packages/website/docs/supported-browsers-and-features.md (hunk 2 above)
Comment (akshay-nm) [reply packages/website/docs/supported-browsers-and-features.md:8]: Its random text 😅

**Comment by akshay-nm** [comment]
> @mvhk Please pull this branch and test out the website as well. I am held up with other projects. I will review the code and merge it if things are working fine.
> 
> @akshay-nm Can you please update a loom file of how it looks like.

I don't know what you are talking about.

**Comment by udaypydi** [comment]
> > @mvhk Please pull this branch and test out the website as well. I am held up with other projects. I will review the code and merge it if things are working fine.
> > @akshay-nm Can you please update a loom file of how it looks like.
> 
> I don't know what you are talking about.

@akshay-nm If the website is functional , it would be great if you can upload a screencast of it.(to the issue ticket). Just for the reference.

**Comment by akshay-nm** [comment]
Here are screens:
Screenshot from 2020-08-03 22-41-24
Screenshot from 2020-08-03 22-41-28
Screenshot from 2020-08-03 22-41-40
Screenshot from 2020-08-03 22-41-46
Screenshot from 2020-08-03 22-41-56
Screenshot from 2020-08-03 22-42-31
Screenshot from 2020-08-03 22-42-48
Screenshot from 2020-08-03 22-42-53
Screenshot from 2020-08-03 22-42-58

@udaypydi

**Comment by akshay-nm** [comment]
This is based on bootstrap. I looked up tailwindcss and I can do that too but I think it would better if we do that in a separate PR.

**Comment by udaypydi** [comment]
Looks great @akshay-nm . Will review the code and merge it. 
Thanks!

**Comment by udaypydi** [comment]
@akshay-nm i have changed the base. Will merge to a different branch. Willl add the packaging part and then merge it to master

**Comment by udaypydi** [comment]
@akshay-nm can you pls resolve the conflicts for one last time?

**Comment by akshay-nm** [comment]
I am afraid that I won't be able to do that. I tried rebasing but there are too many conflicts. 
I might end up breaking things rather than fixing them.
I guess this is a lost cause now.

**Comment by udaypydi** [comment]
@akshay-nm I will give a simpler solution which is not an apt one though. If you haven't made any changes to the `button.component.jsx` then just use the masters code for that component and paste it in yours. For package.json just let your changes prevail. However we are merging to an integration branch.

I won't be able to resolve conflicts in your branch, else I would have done that.

**Comment by akshay-nm** [comment]
I merged master into my branch. I also moved the changes in your package.json to boilerplate package. yarn.lock should also be fine as it was regenerated as per new package.json files.

**Comment by udaypydi** [comment]
@akshay-nm Finally merged it 😄 ! I will add few components and merge it to master.

**Comment by udaypydi** [comment]
if you are interested, you can have a look at the issues section and work on them. 
Thanks for the contribution!

**Pull request merged**